packages/core/snapshot/snapshotread.go

```diff
@@ -0,0 +1,157 @@
+package snapshot
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/cockroachdb/errors"
+	"github.com/iotaledger/goshimmer/packages/core/epoch"
+	"github.com/iotaledger/goshimmer/packages/core/ledger"
+	"github.com/iotaledger/hive.go/serix"
+)
+
+// StreamSnapshotDataFrom consumes a snapshot from the given reader.
+func StreamSnapshotDataFrom(
+	reader io.ReadSeeker,
+	outputConsumer OutputWithMetadataConsumerFunc,
+	epochDiffsConsumer EpochDiffsConsumerFunc,
+	notarizationConsumer NotarizationConsumerFunc) error {
+
+	header, err := ReadSnapshotHeader(reader)
+	if err != nil {
+		return err
+	}
+
+	scanner := bufio.NewScanner(reader)
+	scanner.Split(scanDelimiter)
```

Comment:
Question, how is the size of read batch determined, i.e how much outputs will be read in a single .Bytes() call?